UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs

```diff
@@ -1,106 +1,220 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using UnityEngine;
 using Mirror;
-using Weapons;
+using Weapons.Projectiles;
 
-/// <summary>
-/// Informs all clients that a shot has been performed so they can display it (but they needn't
-/// perform any damage calculation, this is just displaying the shot that the server has already validated).
-/// </summary>
-public class ShootMessage : ServerMessage {
-
-	/// <summary>
-	/// GameObject of the player performing the shot
-	/// </summary>
-	public uint Shooter;
+namespace Weapons
+{
 	/// <summary>
-	/// Weapon being used to perform the shot
+	/// Informs all clients that a shot has been performed so they can display it (but they needn't
+	/// perform any damage calculation, this is just displaying the shot that the server has already validated).
 	/// </summary>
-	public uint Weapon;
-	/// <summary>
-	/// Direction of shot, originating from Shooter)
-	/// </summary>
-	public Vector2 Direction;
-	/// <summary>
-	/// targeted body part
-	/// </summary>
-	public BodyPartType DamageZone;
-	/// <summary>
-	/// If the shot is aimed at the shooter
-	/// </summary>
-	public bool IsSuicideShot;
-
-	///To be run on client
-	public override void Process()
+	public class ShootMessage : ServerMessage
 	{
-		if (!MatrixManager.IsInitialized) return;
 
-		if (Shooter.Equals(NetId.Invalid)) {
-			//Failfast
-			Logger.LogWarning($"Shoot request invalid, processing stopped: {ToString()}", Category.Firearms);
-			return;
+		/// <summary>
+		/// GameObject of the player performing the shot
+		/// </summary>
+		public uint Shooter;
+		/// <summary>
+		/// Weapon being used to perform the shot
+		/// </summary>
+		public uint Projectile;
+		/// <summary>
+		/// Direction of shot, originating from Shooter)
+		/// </summary>
+		public Vector2 Direction;
+		/// <summary>
+		/// targeted body part
+		/// </summary>
+		public BodyPartType DamageZone;
+		/// <summary>
+		/// If the shot is aimed at the shooter
+		/// </summary>
+		public bool IsSuicideShot;
+
+		///To be run on client
+		public override void Process()
+		{
+			if (!MatrixManager.IsInitialized) return;
+
+			if (Shooter.Equals(NetId.Invalid))
+			{
+				//Failfast
+				Logger.LogWarning($"Shoot request invalid, processing stopped: {ToString()}", Category.Firearms);
+				return;
+			}
+
+			//Not even spawned don't show bullets
+			if (PlayerManager.LocalPlayer == null) return;
+
+			LoadMultipleObjects(new uint[] { Shooter, Projectile });
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6363081)

Reply:
how do you feel about this one? I think we can ignore it

Reply:
Oh, already force-pushed. Up to you.